### DISCUSSION TITLE
Multi-process usage

### DISCUSSION THREAD

**Discussion by gertvanantwerpentno** [CESNET/GPUJPEG#68] [Q&A]
What is more efficient: doing all images sequentially in one process, or doing multiple processes simultaneously their own set of images? Can multiple compressions overlap in time?

**Comment by MartinPulec** [comment]
You mean from the time point of view? Definitely from multiple processes (or using multiple threads + CUDA streams) - the usage of GPU is more efficient because computations and memory transfers can interleave.

**Reply by gertvanantwerpentno** [reply]
This seems nice that I/O and computation can overlap, but what is the CPU doing in the mean time? When I write an endless loop, doing nothing more than _gpujpeg_decoder_decode_ followed_ by _gpujpeg_encode_encode_, I see it can compress/decompress 75 times per second with my 1728*3072 images on my machine. But the CPU has also one core running 100% usage. I assume transfer is DMA, so what is the CPU doing?
